environment.sh

```diff
@@ -49,80 +69,101 @@ if [ $env_type == 'simulation' ]; then
 elif [ $env_type == 'artifact' ]; then
   install_file="artifact_requirements.txt"
 else
+  echo
   echo "Invalid environment type. Valid argument types are 'simulation' and 'artifact'."
-  return 
+  exit 1 
 fi
 
 # Pull repo to get latest changes from remote if remote exists
+set +e # Do not exit on error for git ls-remote
 git ls-remote --exit-code --heads origin $branch_name >/dev/null 2>&1
 exit_code=$?
+set -e # Re-enable exit on error
 if [[ $exit_code == '0' ]]; then
   git fetch --all
-  echo "Git branch '$branch_name' exists in the remote repository, pulling latest changes..."
+  echo
+  echo "Git branch '$branch_name' exists in the remote repository; pulling latest changes"
   git pull origin $branch_name
 fi
 
 # Check if environment exists already
-create_env=$(conda info --envs | grep $env_name | head -n 1)
-if [[ $create_env == '' ]]; then
+env_info=$(conda info --envs | grep $env_name | head -n 1)
+if [[ $env_info == '' ]]; then
   # No environment exists with this name
-  echo "Environment $env_name does not exist."
+  echo
+  echo "Environment $env_name does not exist"
   create_env="yes"
   env_exists="no"
 elif [[ $make_new == 'yes' ]]; then
   # User has requested to make a new environment
-  echo "Making a new environment."
+  echo
+  echo "Making a new environment"
   create_env="yes"
   env_exists="yes"
 else
   env_exists="yes"
   conda activate $env_name
   # Check if existing environment needs to be recreated
-  echo "Existing environment found for $env_name."
-  one_week_ago=$(date -d "7 days ago" '+%Y-%m-%d %H:%M:%S')
+  echo
+  echo "Existing environment found for $env_name"
+  expiration_time=$(date -d "$days_until_stale days ago" +%s)
   creation_time="$(head -n1 $CONDA_PREFIX/conda-meta/history)"
   creation_time=$(echo $creation_time | sed -e 's/^==>\ //g' -e 's/\ <==//g')
-  requirements_modification_time="$(date -r $install_file '+%Y-%m-%d %H:%M:%S')"
+  creation_time="$(date -d "$creation_time" +%s)"
+  requirements_modification_time="$(date -r $install_file +%s)"
   # Check if existing environment is older than a week or if environment was built 
   # before last modification to requirements file. If so, mark for recreation.
-  if [[ $one_week_ago > $creation_time ]] | [[ $creation_time < $requirements_modification_time ]]; then
```

Comment:
this never worked for two reasons:
1. The > and < tests were not working correctly because it was trying to compare string representations of dates.
2. A single `|` is shell is pipe, NOT "or" - had to change to `||`